lib/atlas/runner.rb

```diff
@@ -65,24 +50,32 @@ def runtime
       @runtime ||= Runtime.new(dataset, graph)
     end
 
-    #######
     private
-    #######
 
-    # Internal: Given an +element+ from the graph -- a node or edge --
-    # calculate any Rubel queries which are defined on the associated
-    # ActiveDocument.
-    def calculate_rubel_attributes!(element)
-      model = element.get(:model)
+    def refinery_scope
+      dataset.is_a?(Dataset::DerivedDataset) ? :import : :all
+    end
 
-      model.queries && model.queries.each do |attribute, rubel_string|
-        # Skip slot shares.
-        unless attribute.match(/^(?:in|out)put\./)
-          element.set(attribute, query(rubel_string))
-        end
-      end
+    def catalysts_for_refinery_graph(which)
+      catalysts.select { |_, scopes| scopes.include?(which) }.keys
     end
 
+    def catalysts
+      {
+        SetRubelAttributes.with_queryable(method(:query)) =>
+          [:export, :all],
+        Refinery::Catalyst::FromTurbine =>
+          [:import, :export, :all],
```

Comment:
Now that I think about it, for the :import this should even be done before merging in the properties from the YAML file. The properties were exported from Refinery nodes/edges not Turbine nodes/edges.
Maybe the GraphBuilder should output a graph with Refinery nodes already?